manifests/custom_service.pp

```diff
@@ -45,30 +43,25 @@
   Enum['present','absent'] $ensure      = 'present',
 ) {
 
-  include firewalld::reload
+  $_args = delete_undef_values({
+    'short'            => $short,
+    'description'      => $description,
+    'ports'            => $port,
+    'modules'          => $module,
+    'ipv4_destination' => $destination.dig('ipv4'),
+    'ipv6_destination' => $destination.dig('ipv6'),
+  })
 
-  # Service files may only contain alphanumeric characters and underscores.
-  # This is not documented, but has been experimentally confirmed.
+  # Remove legacy files so that we don't end up with conflicts
   $_safe_filename = firewalld::safe_filename($filename)
 
-  $_content = epp(
-    "${module_name}/service.xml.epp",
-    {
-      'short'       => $short,
-      'description' => $description,
-      'port'        => $port,
-      'module'      => $module,
-      'destination' => $destination,
-      'filename'    => $filename,
-      'config_dir'  => $config_dir,
-      'ensure'      => $ensure
+  unless $_safe_filename == $name {
+    file { "${config_dir}/${_safe_filename}.xml":
+      ensure => absent,
     }
-  )
+  }
 
-  file{ "${config_dir}/${_safe_filename}.xml":
-    ensure  => $ensure,
-    content => $_content,
-    mode    => '0644',
-    notify  => Class['firewalld::reload'],
+  firewalld_custom_service { $short:
```

Comment:
Now that `short` is a property in its own right, this should go back to being `$name` (or maybe `$_safe_filename`)?

Reply:
🤦‍♂ Yup, should have been `$_safe_filename`

Reply:
Looks like I transposed those two lines. Will re-test and update